renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -3056,6 +3401,98 @@ void VkResourceRecord::MarkBufferFrameReferenced(VkResourceRecord *buf, VkDevice
     MarkMemoryFrameReferenced(buf->baseResource, buf->memOffset + offset, size, refType);
 }
 
+void VkResourceRecord::MarkBufferImageCopyFrameReferenced(
+    VkResourceRecord *buf, VkResourceRecord *img, const ImageLayouts &layout, uint32_t regionCount,
+    const VkBufferImageCopy *regions, FrameRefType bufRefType, FrameRefType imgRefType)
+{
+  MarkResourceFrameReferenced(img->GetResourceID(), imgRefType);
+  MarkResourceFrameReferenced(img->baseResource, imgRefType);
+
+  // mark buffer just as read
+  MarkResourceFrameReferenced(buf->GetResourceID(), eFrameRef_Read);
+
+  for(uint32_t ri = 0; ri < regionCount; ri++)
+  {
+    const VkBufferImageCopy &region = regions[ri];
+
+    VkFormat regionFormat = layout.format;
+    uint32_t plane = 0;
+    switch(region.imageSubresource.aspectMask)
+    {
+      case VK_IMAGE_ASPECT_STENCIL_BIT: regionFormat = VK_FORMAT_S8_UINT; break;
+      case VK_IMAGE_ASPECT_DEPTH_BIT: regionFormat = GetDepthOnlyFormat(layout.format); break;
+      case VK_IMAGE_ASPECT_PLANE_1_BIT: plane = 1; break;
+      case VK_IMAGE_ASPECT_PLANE_2_BIT: plane = 2; break;
+      default: break;
+    }
+
+    TexelShape texelShape = GetTexelShape(regionFormat, plane);
+
+    // width of copied region, in texels
+    uint32_t widthInTexels = region.imageExtent.width;
```

Comment:
Do you need to divide by the shape's block width here? The image extent is specified in texels AFAIK, so e.g. a 256x256 BC2_UNORM texture would have an image extent here of width=256, height=256. With `texelShape.bytes` of 16 I think this would overestimate `widthInBytes` at the moment.

There's VU guaranteeing that the extent must be a multiple of the block size, so we should be safe to divide and then calculate in blocks, with e.g. `widthInBlocks` etc.

Reply:
Yes. I had implemented this correctly (which is why I wrote a function to get the block shape), and then confused myself on the spec's terminology of `texel` vs `block`, and changed it back. We need the width/height in blocks here, and I think it is now computing this correctly.

Re: the image being a multiple of the block size, I am using the ceiling of the division, as this should safer (in the case of an application not following that valid usage) and negligibly slower than the floor of the division.

Reply:
Yep that's totally fine (doing ceiling rather than floor). As you say it doesn't cost anything, as long as it's not expensive I don't mind being tolerant of VU breakages.